src/components/xcpl_comps_nuopc/xatm/src/atm_comp_nuopc.F90

```diff
@@ -396,6 +411,24 @@ subroutine ModelAdvance(gcomp, rc)
     !--------------------------------
     ! diagnostics
     !--------------------------------
+#ifdef SMARTREDIS
```

Comment:
Can you put this new block of code in its own subroutine? Then the new variable declarations and use statements could also be isolated to that subroutine. That would be especially helpful if you need to keep the ifdefs here: they could be isolated to one specific subroutine. But even if you can get rid of the ifdefs, it could be nice to have this specific code encapsulated in its own subroutine.

Reply:
done